src/app/rpmostree-builtin-kargs.c

```diff
@@ -160,42 +268,91 @@ rpmostree_ex_builtin_kargs (int            argc,
-  const char *existing_kernel_arg_string = NULL;
+  const char *old_kernel_arg_string = NULL;
   if (!g_variant_lookup (boot_config, "options",
-                         "&s", &existing_kernel_arg_string))
+                         "&s", &old_kernel_arg_string))
     return EXIT_FAILURE;
 
   if (display_kernel_args)
     {
-      g_print("The kernel arguments are:\n%s\n",
-              existing_kernel_arg_string);
+      g_print ("The kernel arguments are:\n%s\n",
+               old_kernel_arg_string);
       return EXIT_SUCCESS;
     }
 
   g_autofree char *transaction_address = NULL;
   char *empty_strv[] = {NULL};
 
-  /* dbus does not allow NULL to mean the empty string array,
-   * assign them to be empty string array here to prevent erroring out
-   */
-  if (!opt_kernel_replace_strings)
-    opt_kernel_replace_strings = empty_strv;
-  if (!opt_kernel_append_strings)
-    opt_kernel_append_strings = empty_strv;
-  if (!opt_kernel_delete_strings)
-    opt_kernel_delete_strings = empty_strv;
-
-  /* call the generearted dbus-function */
-  if (!rpmostree_os_call_kernel_args_sync (os_proxy,
-                                           existing_kernel_arg_string,
-                                           (const char* const*) opt_kernel_append_strings,
-                                           (const char* const*) opt_kernel_replace_strings,
-                                           (const char* const*) opt_kernel_delete_strings,
-                                           get_kargs_option_variant (),
-                                           &transaction_address,
+  if (opt_editor)
+    {
+      /* We track the kernel arg instance before the editor */
+      const char *sysroot_path = rpmostree_sysroot_get_path (sysroot_proxy);
+      g_autoptr(GFile) sysroot_file = g_file_new_for_path (sysroot_path);
+      g_autoptr(OstreeSysroot) before_sysroot = ostree_sysroot_new (sysroot_file);
+      if (!ostree_sysroot_load (before_sysroot, cancellable, error))
+        return FALSE;
+
+      const char* current_kernel_arg_string = NULL;
+      if (!kernel_arg_handle_editor (old_kernel_arg_string, &current_kernel_arg_string,
+                                     cancellable, error))
+        return EXIT_FAILURE;
+      gboolean out_changed = FALSE;
+
+      /* Here we load the sysroot again, if the sysroot
+       * has been changed since then, we directly error
+       * out.
+       */
+      if (!ostree_sysroot_load_if_changed (before_sysroot,
+                                           &out_changed,
                                            cancellable,
                                            error))
-    return EXIT_FAILURE;
+        return EXIT_FAILURE;
+      if (out_changed)
+        {
+          g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                       "Conflict: bootloader configuration changed, Saved kernel arguments: \n%s", current_kernel_arg_string);
```

Comment:
This is OK, though it feels cleaner to just `g_print` the saved arguments instead of making it part of the error message? Hmm not sure. At least `s/,/./` though :)

Reply:
sure =).